### ISSUE TITLE
[Bug]: Error: EPERM: operation not permitted, rename on macOS

### ISSUE THREAD

**Issue by jediwade** [electron/electron#31907]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

16.0.1

### What operating system are you using?

macOS

### Operating System Version

macOS Big Sur 11.6.1

### What arch are you using?

arm64 (including Apple Silicon)

### Last Known Working Electron version

n/a

### Expected Behavior

When I use fs.rename(), I expect files to be renamed.

### Actual Behavior

I built a bulk file renamer app in Electron almost 2 years ago for Windows and it works as expected on Windows. I recently started working on porting it to macOS so I could sell it on the Mac App Store, but I can't actually rename any file from a packaged application for MAS. Every file I try (from multiple locations on the computer) end with a "Error: EPERM: operation not permitted, rename" error message. I can successfully rename files if I am just testing the app by launching it with "npm electron ." from Terminal.

I am using the promise version of fs.rename (but it fails on callback version too). As a test, I have added my app to the list of Full Disk Access apps under System Preferences -> Security & Privacy -> Privacy -> Full Disk Access, but this makes no difference.

Strangely, I am able to read and create files from my app in any directory on the computer without my app being in the list of Full Disk Access apps.

I am never prompted by macOS to grant access to any folder in order to rename, read, or create files.

My entitlements.mas.plist file contains "com.apple.security.app-sandbox", "com.apple.security.files.user-selected.read-only", and "com.apple.security.files.user-selected.read-write" with all set to true. The "gatekeeperAssess" and "hardenedRuntime" are set to false. I package my app with electron-builder. I am running the latest (17.1.0) Node as downloaded and installed from the NodeJS website.

**Comment by jediwade** [comment]
I just tried renameSync and get the same error message. I also found out that I can successfully rename a file if the path (new folder location) is changed. It seems the problem is strictly renaming a file in place.

**Comment by jediwade** [comment]
Just tried to use the ChildProcess.exec instead of fsPromise.rename() to run "mv test.txt asdtest.txt" and got an error with that as well.

Code used:
const _exec = require('child_process').exec;
_exec("mv " + oldpath + " " + newpath, {windowsHide:true} , function(error, stdout, stderr) {
	console.log(error);
	console.log(stdout);
	console.log(stderr);
});

Error message below:
﻿
VM114 preload.js:111
Error: Command failed: mv /Users/jediwade/Desktop/test.txt /Users/jediwade/Desktop/asdtest.txt
mv: rename /Users/jediwade/Desktop/test.txt to /Users/jediwade/Desktop/asdtest.txt: Operation not permitted
    at ChildProcess.exithandler (VM126 node:child_process:406)
    at ChildProcess.emit (VM14 node:events:394)
    at maybeClose (VM127 child_process:1064)
    at Socket.<anonymous> (VM127 child_process:450)
    at Socket.emit (VM14 node:events:394)
    at Pipe.<anonymous> (VM128 node:net:672)
VM114 preload.js:112 
(empty output for stdout)

VM114 preload.js:113
mv: rename /Users/jediwade/Desktop/test.txt to /Users/jediwade/Desktop/asdtest.txt: Operation not permitted

**Comment by jediwade** [comment]
I got hold of an Intel MacBook Pro and got the same results when using a packaged/signed app.

**Comment by jediwade** [comment]
This is interesting. If I drag and drop folders into my app, I can rename files. But if I drag-n-drop files directly, I get the EPERM error. I updated my code so that files get handled in the same function that folders do and it did not change any of the results. There seems to be something different about dropping folders vs dropping individual files and selecting individual files with an HTML file input element.

**Comment by jediwade** [comment]
I downgraded my Electron version to v12.2.3 as a test to see if this still occurred on older versions and it does.

**Comment by jediwade** [comment]
If I execute window.showDirectoryPicker() and select a folder with that prompt, any file I add (through drag-n-drop or browsing the file system with an HTML file input) to be renamed works for that folder and as long as the app is running.

**Comment by jediwade** [comment]
Just tried renaming a file that I knew the path to under main.js and it still failed as a packaged app.

**Comment by jediwade** [comment]
It appears I can resolve this by making use of the security scoped bookmarks through dialog.showOpenDialog() and saving the string generated from the selected folder, but I would argue that this is not a great option because I still cannot have my app present the "X.app wants to access your Documents folder [Deny] [OK]" prompt and I have to put in a lot more work just to make my app work for a Mac App Store app.

**Comment by jediwade** [comment]
Ran another test with the latest version of Electron (18.0.4) and this problem still exists (temporarily removed security scoped bookmarks for testing). I did experience something I believe is new. When I try to rename files added through file browser, it fails (as expected). But when I try to rename the files again in the same app session, it works and continues to work for any file in any folder.

MacBook Air M1 is now running macOS Monterey 12.2.1 and I am now only building x64.
